contracts/Dragon/DragonLeaderboard.sol

```diff
@@ -82,9 +82,11 @@ contract DragonLeaderboard is Upgradable {
         lastRewardDate = now; // solium-disable-line security/no-block-members
     }
 
-    function getRewards(uint256 _hatchingPrice) external view returns (uint256[10] rewards) {
-        for (uint8 i = 0; i < 10; i++) {
-            rewards[i] = _hatchingPrice.mul(rewardCoefficients[i]).div(COEFFS_MULTIPLIER);
+    function getRewards(uint256 _remainingGold) external view returns (uint256[10] rewards) {
+        for (uint8 i = 0; i < REWARDED_DRAGONS_AMOUNT; i++) {
+            rewards[i] = _remainingGold.mul(uint256(2).pow(REWARDED_DRAGONS_AMOUNT.sub(1))).div(
```

Comment:
I'm not sure that the formula is correct, because after the treasury balance will be less than `10,230,000` the last 10th user will start receiving less than 1 gold. @xhipster could you check it please?

Reply:
Its expected that Leaderboard rewards will be slowed down significantly.